services/libs/aster-std/src/process/vm/mod.rs

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! This module defines struct `Vm` to represent the layout of user space process virtual memory.
+//!
+//! The `Vm` struct contains `Vmar`, which stores all existing memory mappings. The `Vm` also contains
+//! the basic info of process level vm segments, like init stack and heap.
+//!
+//! TODO: The vm layout should be redesigned in the future. Currently, we only use a small part of the
+//! whole virtual memory space.
+//!
+
+mod heap;
+mod stack;
+
+use super::constants::{MAX_ARG_LEN, MAX_ENVP_NUMBER, MAX_ENV_LEN};
+use crate::prelude::*;
+use crate::vm::vmar::Vmar;
+use alloc::ffi::CString;
+use aster_frame::vm::VmIo;
+use aster_rights::Full;
+
+pub use heap::Heap;
+pub use stack::Stack;
+
+/*
+* The user space virtual memory layout looks like below.
+* |-----------------------|-------The highest user vm address
+* |                       |
+* |       Mmap Areas      |
+* |                       |
+* |                       |
+* --------------------------------The init stack base
+* |                       |
+* | User Stack(Init Stack)|
+* |                       |
+* |         ||            |
+* ----------||----------------------The user stack top, grows down
+* |         \/            |
+* |                       |
+* |     Unmapped Areas    |
+* |                       |
+* |         /\            |
+* ----------||---------------------The user heap top, grows up
+* |         ||            |
+* |                       |
+* |        User Heap      |
+* |                       |
+* ----------------------------------The user heap base
+*/
+
+/// The process user space virtual memory
+pub struct Vm {
```

Comment:
You really think `Vm` is better than `ProcessVm`? I am worried that the name `Vm` is too vague. There are too many stuff related to virtual memory. Calling this struct `Vm` instead of `ProcessVm` could be confusing.